### PR TITLE
docs: add CLAUDE.md behavioral guide

### PR DESCRIPTION
**Pull request by ankita10119**
### 🔧 Changes

Add `CLAUDE.md` and `PR_REVIEW_GUIDE.md` to encode development and review standards directly into the repo.                                            
                                                                                                                                                         
**CLAUDE.md** - behavioral guide for working in this repo (for both humans and AI agents):
  - Feature development protocol: step-by-step process for implementing handlers or API specs, including pre-code questions to answer, the four required handler methods (`getType`, `calcChanges`, `processChanges`, `validate`), test requirements, and format verification                                   
  - Pre-submit gate: four questions that must be answered before marking work complete, plus required local checks                                       
  - deploy-cli invariants: non-negotiable constraints for every handler/context parser change (both formats, dry-run, keyword replacement, write-only/read-only field stripping, identifier matching, ordering, schema)
  - EA / entitlement-gated feature rules: how to document untested code paths and API assumptions when a feature requires special entitlement to test end-to-end
  - Traps: high-signal patterns from real bugs in this codebase

**PR_REVIEW_GUIDE.md** - living checklist for thorough PR reviews, covering 13 areas: context, correctness, edge cases, backward compatibility, safety, tests, docs, code quality, performance, security, CI, deploy-cli specifics, and EA/entitlement-gated scenarios. Includes final sign-off questions and lessons learned from real reviews.

### 📚 References

N/A

### 🔬 Testing

No functional code changes - documentation only. Verify by reading both files and checking that the deploy-cli specific claims match the codebase (handler methods, keyword replacement patterns, `stripCreateFields`/`stripUpdateFields`, `@order()` usage).

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [ ] I have added documentation for all new/changed functionality (or N/A)

### CODE REVIEW

File: references/code-style.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+```typescript
+// ✅ Stable name-based identifier
+static identifiers = ['name'];
+
+// ✅ ID alongside name (when the API uses both for matching)
+static identifiers = ['client_id', 'name'];
+
+// 🚫 Never — sole auto-generated UUID breaks cross-tenant portability
+static identifiers = ['id'];
+```
```
Comment (harshithRai) [inline references/code-style.md:48]: Nit: the other snippets in this file are real idioms, but `static identifiers` isn't used anywhere in `src/`. The base class reads `options.identifiers` passed to `super()` (see `clients.ts:691`), so this exact form would no-op if copied. Might be worth matching the real syntax so it's not mistaken for copy-paste code:

​```typescript
// in constructor options passed to super()
identifiers: ['client_id', 'name'],
​```

**Review by harshithRai — COMMENTED** [review]
Just a small nit suggestion below. LGTM otherwise, pending the format check fix

File: references/code-style.md (hunk 1 above)
Comment (ankita10119) [reply references/code-style.md:48]: Addressed and updated

**Review by harshithRai — DISMISSED** [review]

**Review by harshithRai — APPROVED** [review]
